addons/mail/models/discuss/res_partner.py

```diff
@@ -94,7 +94,7 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
             *self.env["discuss.channel.member"]._to_store_persona([]),
         ]
         store = Store(members, member_fields).add(partners)
-        store.add(channel, {"group_public_id": channel.group_public_id.id if channel.group_public_id else None})
+        store.add(channel, {"group_public_id": channel.group_public_id.id} if channel.group_public_id else None)
```

Comment:
Isn't this just a complex way to write?
```suggestion
        store.add(channel, "group_public_id")
```

Reply:
I don't understand the issue, the field value should always be correct. It's acceptable (and what we want) to send an empty value if the field is empty.

Reply:
I think with just `store.add(channel, "group_public_id")` it should be fine.

The issue was that is previously returned an empty recordset instead of a falsy value right?

Reply:
Do we need to fix that in earlier versions of just master?

Reply:
You might already make the same fix here so we don't have to wait for the forward-port.